### ISSUE TITLE
should pass node options or v8 options to the  subcommend exec process

### ISSUE THREAD

**Issue by fishbar** [tj/commander.js#877]
```js
Command.prototype.executeSubCommand = function(argv, args, unknown) {
```
when subcommand executing, the unknow args will passed to the sub process which spawn by the function above. but if the args is belong to node（

node options or v8 options）, it will fail。

for example, when the subcommand deal with something which needs large mem, so we need to setup the v8 options  `--max_old_space_size =2048`,  for now the `max_old_space_size ` will be treated 
as unknow options, so the process which running the subCommand can not receive these options.

**Comment by fishbar** [comment]
or these node options can be passed by ENV `COMMANDER_NODE_OPTIONS`

**Comment by vanesyan** [comment]
Hello there can you please elaborate on it, I'm sure I have correctly understood you

**Comment by fishbar** [comment]
for example，when the command running with large memory and deep stacks， i need pass v8 options to the node process，like：

```sh 
> customCmd  subcmd --max_old_space_size=2048 --stack_size=1024  other_options
```

now commander will treated these v8 options as unknow arguments and passed to the subcmd。
so the v8 options dose’t work， and because of the memory limitation and stack limitation，command never can success.

**Comment by shadowspawn** [comment]
@fishbar 
Can you use `NODE_OPTIONS`?

https://nodejs.org/api/cli.html#cli_node_options_options

**Comment by shadowspawn** [comment]
@fishbar are you on Windows?

(There is code passing the extra arguments for javascript subcommands when not on Windows.)

**Comment by shadowspawn** [comment]
This will hopefully be fixed by #611

**Comment by shadowspawn** [comment]
This issue will be resolved when v3.0.0 is released. Available now as a prerelease. See #1001

**Comment by shadowspawn** [comment]
Support for passing node options added in v3 (thanks @hxsf): https://github.com/tj/commander.js/releases/tag/v3.0.0
